app/scripts/models/brand.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+var Brand = (function () {
+
+  var Brand = function Brand(data) {
+    this.categoryIdsCsl = data.hasOwnProperty('categories') ? data.categories : null;
+    this.description = data.hasOwnProperty('description') ? data.description : null;
+    this.id = data.hasOwnProperty('id') ? parseInt(data.id) : null;
+    this.image = data.hasOwnProperty('image') ? data.image : null;
+    this.name = data.hasOwnProperty('name') ? data.name : null;
+    this.ranking = data.hasOwnProperty('ranking') ? data.ranking : null;
+    this.brandUrl = null; // @TODO: There does not seem to be a URL property in the data.
+
+    // Convert the category ids CSL to an array and remember it.
+    this.categoryIds = this.categoryIdsCsl === null ? [] : String(this.categoryIdsCsl).split(',');
+  };
+
+  Brand.prototype = {
+    getId: function getId() {
+      return this.id;
+    },
+
+    hasId: function hasId() {
+      return this.id !== null;
+    },
+
+    setId: function setId(newId) {
+      this.id = newId;
+    },
+
+    hasName: function hasName() {
+      return this.name !== null;
+    },
+
+    getName: function getName() {
+      return this.name;
+    },
+
+    setName: function setName(name) {
+      this.name = name;
+    },
+
+    hasImage: function hasImage() {
+      return this.image !== null;
+    },
+
+    getImage: function getImage() {
+      return this.image;
+    },
+
+    setImage: function setImage(imageData) {
+      this.image = imageData;
+    },
+
+    isImageAnUpload: function isImageAnUpload() {
```

Comment:
`isImageAnUpload` is a confusing name

Reply:
I really struggled to think of a good name for this method, and obviously the one I came up with wasn't good enough!

I'm still struggling to think of a good name. The best I could come up with is "imageRequiresUploading", and I've changed it to that in all the models.

If you can think of a better name, feel free to change it :-)